samples/extensions/ray_tracing_position_fetch/ray_tracing_position_fetch.cpp

```diff
@@ -0,0 +1,619 @@
+	raytracing_pipeline_create_info.maxPipelineRayRecursionDepth = 1;
+	raytracing_pipeline_create_info.layout                       = pipeline_layout;
+	VK_CHECK(vkCreateRayTracingPipelinesKHR(get_device().get_handle(), VK_NULL_HANDLE, VK_NULL_HANDLE, 1, &raytracing_pipeline_create_info, nullptr, &pipeline));
+}
+
+/*
+    Create the uniform buffer used to pass matrices to the ray tracing ray generation shader
+*/
+void RayTracingPositionFetch::create_uniform_buffer()
+{
+	ubo = std::make_unique<vkb::core::Buffer>(get_device(), sizeof(uniform_data), VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+	ubo->convert_and_update(uniform_data);
+	update_uniform_buffers();
+}
+
+/*
+    Command buffer generation
+*/
+void RayTracingPositionFetch::build_command_buffers()
+{
+	if (width != storage_image.width || height != storage_image.height)
+	{
+		// If the view port size has changed, we need to recreate the storage image
+		vkDestroyImageView(get_device().get_handle(), storage_image.view, nullptr);
+		vkDestroyImage(get_device().get_handle(), storage_image.image, nullptr);
+		vkFreeMemory(get_device().get_handle(), storage_image.memory, nullptr);
+		create_storage_image();
+		// The descriptor also needs to be updated to reference the new image
+		VkDescriptorImageInfo image_descriptor{};
+		image_descriptor.imageView              = storage_image.view;
+		image_descriptor.imageLayout            = VK_IMAGE_LAYOUT_GENERAL;
+		VkWriteDescriptorSet result_image_write = vkb::initializers::write_descriptor_set(descriptor_set, VK_DESCRIPTOR_TYPE_STORAGE_IMAGE, 1, &image_descriptor);
+		vkUpdateDescriptorSets(get_device().get_handle(), 1, &result_image_write, 0, VK_NULL_HANDLE);
+	}
+
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	VkImageSubresourceRange subresource_range = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+
+	for (int32_t i = 0; i < draw_cmd_buffers.size(); ++i)
+	{
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+
+		/*
+		    Setup the strided device address regions pointing at the shader identifiers in the shader binding table
+		*/
+
+		const uint32_t handle_size_aligned = aligned_size(ray_tracing_pipeline_properties.shaderGroupHandleSize, ray_tracing_pipeline_properties.shaderGroupHandleAlignment);
+
+		VkStridedDeviceAddressRegionKHR raygen_shader_sbt_entry{
+		    raygen_shader_binding_table->get_device_address(),
+		    handle_size_aligned,
+		    handle_size_aligned};
+
+		VkStridedDeviceAddressRegionKHR miss_shader_sbt_entry{
+		    miss_shader_binding_table->get_device_address(),
+		    handle_size_aligned,
+		    handle_size_aligned};
+
+		VkStridedDeviceAddressRegionKHR hit_shader_sbt_entry{
+		    hit_shader_binding_table->get_device_address(),
+		    handle_size_aligned,
+		    handle_size_aligned};
+
+		VkStridedDeviceAddressRegionKHR callable_shader_sbt_entry{};
```

Comment:
Aren't those four `VkStridedDeviceAddressRegionKHR` loop invariant? Could then be moved out of the loop.

Reply:
Right, I copied that part from other samples and indeed, it makes no sense to have them inside the loop. Also made them const.